### PR TITLE
bump: Pravega 0.10.1 (was 0.9.1)

### PR DESCRIPTION
**Pull request by cheleb**
KVTable Beta 2, key family removed.
 
Api changes intended.

### CODE REVIEW

File: pravega/src/main/scala/akka/stream/alpakka/pravega/TableEntry.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.pravega
+
+import io.pravega.client.tables.TableKey
+import io.pravega.client.tables.Version
+
+class TableEntry[+V](val tableKey: TableKey, version: Version, val value: V)
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/TableEntry.scala:10]: Can this be `final`?
Why is `version` not a `val`?

File: pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala [hunk 2]
```diff
@@ -365,22 +377,31 @@ object TableReaderSettingsBuilder {
    * Create reader settings from the default configuration
    * `akka.alpakka.pravega`.
    */
-  def apply[K, V](actorSystem: ActorSystem): TableReaderSettingsBuilder[K, V] =
-    apply(actorSystem.settings.config.getConfig(configPath))
+  def apply[K, V](implicit actorSystem: ActorSystem,
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala:380]:
```suggestion
  def apply[K, V]()(implicit actorSystem: ActorSystem,
```

File: pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala [hunk 3]
```diff
@@ -365,22 +377,31 @@ object TableReaderSettingsBuilder {
...
+                  valueSerializer: Serializer[V]): TableReaderSettingsBuilder[K, V] =
+    apply(actorSystem.settings.config.getConfig(configPath), keySerializer, valueSerializer)
 
   /**
    * Create reader settings from the default configuration
    * `akka.alpakka.pravega`.
    */
-  def create[K, V](actorSystem: ActorSystem): TableReaderSettingsBuilder[K, V] =
-    apply(actorSystem)
+  def create[K, V](implicit actorSystem: ActorSystem,
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala:389]:
```suggestion
  def create[K, V](actorSystem: ActorSystem,
```
This is for the Java API which shouldn't use `implicit`.

File: pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala [hunk 4]
```diff
@@ -475,22 +505,31 @@ object TableWriterSettingsBuilder {
    * Create writer settings from the default configuration
    * `akka.alpakka.pravega`.
    */
-  def apply[K, V](actorSystem: ActorSystem): TableWriterSettingsBuilder[K, V] =
-    apply(actorSystem.settings.config.getConfig(configPath))
+  def apply[K, V](implicit actorSystem: ActorSystem,
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala:508]:
```suggestion
  def apply[K, V]()(implicit actorSystem: ActorSystem,
```

File: pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala [hunk 5]
```diff
@@ -475,22 +505,31 @@ object TableWriterSettingsBuilder {
...
+                  valueSerializer: Serializer[V]): TableWriterSettingsBuilder[K, V] =
+    apply(actorSystem.settings.config.getConfig(configPath), keySerializer, valueSerializer)
 
   /**
    * Create writer settings from the default configuration
    * `akka.alpakka.pravega`.
    */
-  def create[K, V](actorSystem: ActorSystem): TableWriterSettingsBuilder[K, V] =
-    apply(actorSystem)
+  def create[K, V](implicit actorSystem: ActorSystem,
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/PravegaSettings.scala:517]:
```suggestion
  def create[K, V](actorSystem: ActorSystem,
```

File: pravega/src/main/scala/akka/stream/alpakka/pravega/scaladsl/PravegaTable.scala [hunk 6]
```diff
@@ -23,61 +24,40 @@ object PravegaTable {
...
-   * Keys, values and key families are extracted from incoming messages and written to Pravega table.
+   * Keys and values are extracted from incoming messages and written to Pravega table.
    * Messages are emitted downstream unchanged.
    */
-  def writeFlow[K, V](scope: String, tableName: String, familyExtractor: K => String)(
-      implicit tableWriterSettings: TableWriterSettings[K, V]
-  ): Flow[(K, V), (K, V), NotUsed] =
+  def writeFlow[K, V](scope: String,
+                      tableName: String,
+                      tableWriterSettings: TableWriterSettings[K, V]): Flow[(K, V), (K, V), NotUsed] =
```
Comment (ennru) [inline pravega/src/main/scala/akka/stream/alpakka/pravega/scaladsl/PravegaTable.scala:50]: The setting were `implicit` earlier. Is the API better with explicit settings here?
I'm OK with what you believe works best.

File: pravega/src/test/scala/docs/scaladsl/PravegaReadWriteDocs.scala [hunk 7]
```diff
@@ -21,14 +21,17 @@ import akka.stream.alpakka.pravega.TableReaderSettingsBuilder
...
 import akka.stream.alpakka.pravega.scaladsl.Pravega
 import scala.util.Using
+import io.pravega.client.tables.TableKey
 
 class PravegaReadWriteDocs {
 
   implicit val system = ActorSystem("PravegaDocs")
 
-  val serializer = new UTF8StringSerializer
+  implicit val serializer = new UTF8StringSerializer
```
Comment (ennru) [inline pravega/src/test/scala/docs/scaladsl/PravegaReadWriteDocs.scala:30]: Prefer to have types on implicits.

**Review by ennru — COMMENTED** [review]
Thank you for keeping it up.
With Alpakka 4.0 coming up we can bring in the API changes.

More places might benefit from adding explicit `()`.

File: pravega/src/main/scala/akka/stream/alpakka/pravega/scaladsl/PravegaTable.scala (hunk 6 above)
Comment (cheleb) [reply pravega/src/main/scala/akka/stream/alpakka/pravega/scaladsl/PravegaTable.scala:50]: Yes, that is not very convenient when multiple settings are in the same scope.

File: pravega/src/test/scala/akka/stream/alpakka/pravega/package.scala [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+ * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka
+
+import io.pravega.client.stream.Serializer
+import java.nio.ByteBuffer
+import io.pravega.client.stream.impl.UTF8StringSerializer
+
+package object pravega {
```
Comment (ennru) [inline pravega/src/test/scala/akka/stream/alpakka/pravega/package.scala:11]: Wouldn't it be more obvious how the serializers get in scope if this was in `package docs.scaladsl` and the a regular `object Serializers`?
By importing from it, the serializers in `PravegaReadWriteDocs` and `PravegaSettingsSpec` could be removed, as well.

File: pravega/src/test/scala/docs/scaladsl/PravegaReadWriteDocs.scala [hunk 9]
```diff
@@ -84,36 +89,37 @@ class PravegaReadWriteDocs {
 
     }
 
-  case class Person(id: String, firstname: String)
+  case class Person(id: Int, firstname: String)
```
Comment (ennru) [inline pravega/src/test/scala/docs/scaladsl/PravegaReadWriteDocs.scala:92]: This class exists in `package akka.stream.alpakka.pravega`, as well. Might be easier with just one?

**Review by ennru — COMMENTED** [review]
I looked more at the tests.
The structure could be simpler, maybe all tests could go in `docs.scaladsl`?

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you!

**Pull request merged**